apps/server/src/modules/provisioning/service/provisioning.service.ts

```diff
@@ -0,0 +1,55 @@
+import { Injectable, InternalServerErrorException, UnprocessableEntityException } from '@nestjs/common';
+import { SystemProvisioningStrategy } from '@shared/domain/interface/system-provisioning.strategy';
+import { Logger } from '@src/core/logger';
+import { ProvisioningSystemInputDto } from '@src/modules/provisioning/dto/provisioning-system-input.dto';
+import { ProvisioningSystemInputMapper } from '@src/modules/provisioning/mapper/provisioning-system-input.mapper';
+import { SanisProvisioningStrategy, SanisStrategyData } from '@src/modules/provisioning/strategy/sanis/sanis.strategy';
+import { IservProvisioningStrategy, IservStrategyData } from '@src/modules/provisioning/strategy/iserv/iserv.strategy';
+import { ProvisioningDto } from '@src/modules/provisioning/dto/provisioning.dto';
+import { SystemService } from '@src/modules/system/service/system.service';
+
+@Injectable()
+export class ProvisioningService {
+	constructor(
+		private readonly systemService: SystemService,
+		private readonly sanisStrategy: SanisProvisioningStrategy,
+		private readonly iservStrategy: IservProvisioningStrategy,
+		private readonly logger: Logger
+	) {
+		this.logger.setContext(ProvisioningService.name);
+	}
+
+	async process(accessToken: string, idToken: string, systemId: string): Promise<ProvisioningDto> {
+		let system: ProvisioningSystemInputDto;
+		try {
+			system = ProvisioningSystemInputMapper.mapToInternal(await this.systemService.findById(systemId));
+		} catch (e) {
+			this.logger.error(`System with id ${systemId} was not found.`);
+			throw new UnprocessableEntityException(`System with id "${systemId}" does not exist.`);
+		}
+
+		switch (system.provisioningStrategy) {
+			case SystemProvisioningStrategy.SANIS: {
+				if (!system.provisioningUrl) {
+					throw new UnprocessableEntityException(`Sanis system with id: ${systemId} is missing a provisioning url`);
+				}
+
+				const params: SanisStrategyData = {
+					provisioningUrl: system.provisioningUrl,
+					accessToken,
+					systemId,
+				};
+				return this.sanisStrategy.apply(params);
+			}
+			case SystemProvisioningStrategy.ISERV: {
+				const params: IservStrategyData = {
+					idToken,
+				};
+				return this.iservStrategy.apply(params);
+			}
+			default:
+				this.logger.error(`Missing provisioning strategy for system with id ${systemId}`);
```

Comment:
The error is logged over the error pipline that handle the throw and log it. You must have trusth that the error pipline filter all crititcal informations. In this case the systemId in error message should be avoid and pass as additional informaton. 
Please let check together if the nestjs errors in combination with our error pipline support this.

Reply:
Allright. I removed the logger completely

Reply:
On this point i also must be looked to our current implementation.